src/lib.rs

```diff
@@ -1,8 +1,60 @@
 //! A Rust implementation of the [GeoArrow](https://github.com/geoarrow/geoarrow) specification,
-//! plus algorithms implemented on and returning these GeoArrow arrays.
+//! including algorithms implemented on and returning these GeoArrow arrays.
+//!
+//! # Reading and writing
+//!
+//! The [io] module has functions for reading and writing GeoArrow data from a variety of formats.
+//! To use most format readers and writers, you must enable their corresponding feature.
+//! For example, to convert between [geosjon](https://geojson.org/) and GeoArrow, enable the `geozero` feature in your `Cargo.toml`:
+//!
+//! ```toml
+//! #[dependencies]
+//! geoarrow = { version = "*", features = ["geozero"] }
+//! ```
+//!
+//! Then:
+//!
+//! ```
+//! # #[cfg(feature = "geozero")]
+//! # {
+//! use std::{io::Cursor, fs::File};
+//!
+//! // Reads geojson from a file into a GeoArrow table.
+//! let file = File::open("fixtures/roads.geojson").unwrap();
+//! let table = geoarrow::io::geojson::read_geojson(file, None).unwrap();
+//!
+//! // Writes that table to a cursor as JSON, then reads it back into a `serde_json::Value`.
+//! let mut cursor = Cursor::new(Vec::new());
+//! geoarrow::io::geojson::write_geojson(table, &mut cursor);
+//! let value: serde_json::Value = serde_json::from_slice(&cursor.into_inner()).unwrap();
+//! # }
+//! ```
+//!
+//! See the [io] module for more information on the available formats and their features.
+//!
+//! # Constructing
+//!
+//! You can build GeoArrow arrays all at once from [mod@geo] structures:
```

Comment:
Maybe also note somewhere here

> You can build GeoArrow arrays from a vector of anything that implements geometry traits. Along with the GeoRust community, geoarrow-rs has been prototyping [geometry access traits](https://github.com/georust/geo/discussions/838) for a standardized way to access coordinate information, regardless of the storage format of the geometries. For now, we vendor an implementation of geo-traits (see [mod@geo_traits]), but this may be upstreamed to georust in the future.

So here this `PointArray::from` implementation works on a `Vec<geo::Point>` because `geo::Point` implements `PointTrait`, but it _also_ works on other things that implement `PointTrait`, such as WKB points or GEOS points or (in the future) FlatGeobuf points. These traits can be implemented on anything where you can provide zero-copy access to individual coordinates.

See also further discussion in https://github.com/georust/geo/pull/1157

Reply:
Excellent, added a note. To that point, should we _not_ add more docs to `geoarrow::geo_traits`, if the plan is to eventually switch from that vendored code to the `geo_traits` crate?

Reply:
I think it's fine to document geo_traits more if you'd like. Our geo_traits will most likely form the basis for an upstream geo_traits module, but we're waiting for things to stabilize a bit (and for consensus). E.g. I just updated the traits to support 3d geometries, and haven't built out things around that yet.